spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesBootstrapConfiguration.java

```diff
@@ -16,16 +16,69 @@
 
 package org.springframework.cloud.kubernetes.commons.config;
 
+import org.aspectj.lang.annotation.Aspect;
+
+import org.springframework.boot.autoconfigure.aop.AopAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.kubernetes.commons.ConditionalOnKubernetesEnabled;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.retry.interceptor.RetryInterceptorBuilder;
+import org.springframework.retry.interceptor.RetryOperationsInterceptor;
+import org.springframework.retry.policy.NeverRetryPolicy;
 
 /**
  * @author Ryan Baxter
+ * @author Isik Erhan
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnKubernetesEnabled
 @EnableConfigurationProperties({ ConfigMapConfigProperties.class, SecretsConfigProperties.class })
 public class KubernetesBootstrapConfiguration {
 
+	@ConditionalOnKubernetesConfigPropertiesRetryEnabled
+	@ConditionalOnClass({ Retryable.class, Aspect.class, AopAutoConfiguration.class })
+	@Configuration(proxyBeanMethods = false)
+	@EnableRetry(proxyTargetClass = true)
+	@Import(AopAutoConfiguration.class)
+	static class RetryConfiguration {
+
+		private static RetryOperationsInterceptor retryOperationsInterceptor(
+				AbstractConfigProperties.RetryProperties retryProperties) {
+			return RetryInterceptorBuilder.stateless().backOffOptions(retryProperties.getInitialInterval(),
+					retryProperties.getMultiplier(), retryProperties.getMaxInterval())
+					.maxAttempts(retryProperties.getMaxAttempts()).build();
+		}
+
+		@Bean
+		@ConditionalOnConfigMapPropertiesRetryEnabled
+		public RetryOperationsInterceptor configMapPropertiesRetryInterceptor(
+				ConfigMapConfigProperties configProperties) {
+			return retryOperationsInterceptor(configProperties.getRetry());
+		}
+
+		@Bean("configMapPropertiesRetryInterceptor")
+		@ConditionalOnConfigMapPropertiesRetryDisabled
+		public RetryOperationsInterceptor configMapPropertiesRetryInterceptorNoRetry() {
+			return RetryInterceptorBuilder.stateless().retryPolicy(new NeverRetryPolicy()).build();
+		}
+
+		@Bean
+		@ConditionalOnSecretsPropertiesRetryEnabled
+		public RetryOperationsInterceptor secretsPropertiesRetryInterceptor(SecretsConfigProperties configProperties) {
+			return retryOperationsInterceptor(configProperties.getRetry());
+		}
+
+		@Bean("secretsPropertiesRetryInterceptor")
```

Comment:
I did not understand initially why you have `@Bean("secretsPropertiesRetryInterceptor")`, but darn this is interesting. I don't know why but I would have expected for `@Retryable` to take an array of `interceptors`? And take the first one present in context? This way, you could have had two clean beans here:

```

		@Bean
		@ConditionalOnSecretsPropertiesRetryEnabled
		public RetryOperationsInterceptor secretsPropertiesRetryInterceptor(SecretsConfigProperties configProperties) {
			return retryOperationsInterceptor(configProperties.getRetry());
		}

		@Bean
		@ConditionalOnSecretsPropertiesRetryDisabled
		public RetryOperationsInterceptor secretsPropertiesNoRetryInterceptor() {
			return RetryInterceptorBuilder.stateless().retryPolicy(new NeverRetryPolicy()).build();
		}
```

and `@Retryable(interceptors = { "secretsPropertiesRetryInterceptor", "secretsPropertiesNoRetryInterceptor" })`. 

This is pure theoretical and I do not know what I am missing, but looking at this hack here, I would have wanted that to be a reality.

Reply:
Unfortunately, `@Retryable` doesn't work like that. It takes a single `interceptor` bean name and tries to get the bean with that name from the bean factory if the bean name is not empty. And if the bean itself is not present in the bean factory, then it fails with `NoSuchBeanDefinitionException`. (see [AnnotationAwareRetryOperationsInterceptor.java](https://github.com/spring-projects/spring-retry/blob/main/src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java#L185-L188)) I'd expect it to work normally without retrying in this case but apparently, it doesn't work like that.

Reply:
or like that, yes. Either way, this hack you made here is nice, thank you for the follow-up here.